CorsixTH/Lua/graphics.lua

```diff
@@ -120,23 +120,37 @@ end
 --! Tries to load the font file given in the config file as unicode_font.
 --! If it is not found it tries to find one in the operating system.
 function Graphics:loadFontFile()
+  local lfs = require("lfs")
+  local function check(path) return path and lfs.attributes(path, "mode") == "file" end
   -- Load the Unicode font, if there is one specified.
-  local font_file = self.app.config.unicode_font
-  if not font_file then
-    -- Try a font which commonly comes with the operating system.
-    local windir = os.getenv("WINDIR")
-    if windir and windir ~= "" then
-      font_file = windir .. pathsep .. "Fonts" .. pathsep .. "ARIALUNI.TTF"
-    elseif self.app.os == "macos" then
-      font_file = "/Library/Fonts/Arial Unicode.ttf"
-    else
-      font_file = "/usr/share/fonts/truetype/arphic/uming.ttc"
-    end
+  local config_path = self.app.config.unicode_font
+  -- Try a font that commonly comes with the operating system.
+  local os_path, font_file
+  local windir = os.getenv("WINDIR")
+  if windir and windir ~= "" then
+    os_path = windir .. pathsep .. "Fonts" .. pathsep .. "ARIALUNI.TTF"
+  elseif self.app.os == "macos" then
+    os_path = "/Library/Fonts/Arial Unicode.ttf"
+  else
+    os_path = "/usr/share/fonts/truetype/arphic/uming.ttc"
+  end
+  if check(config_path) then font_file = config_path
+  elseif check(os_path) then
+    font_file = os_path
+    print("Configured unicode font not found, using " .. font_file .. " instead.")
+    print("This will be written to the config file.")
+  elseif config_path ~= nil then
+    print("Configured unicode font not found, no fallback available.")
```

Comment:
We still have the game font though, right?

Reply:
Yes, this is preparation for the player to pick a unicode language. That line could say that only the non-unicode languages are available?

Reply:
I think we'll need to bear in mind #1914 and #1915 . How do you see this as a larger plan of attack of those? Open question.

I'm not against this pr though, don't get me wrong.

Reply:
Maybe this is a misstep towards the greater solution described in #1901